src/components/claims/claim-details/components/DeleteClaimFileButton.tsx

```diff
@@ -0,0 +1,109 @@
+import gql from 'graphql-tag'
+import * as React from 'react'
+import { Mutation } from 'react-apollo'
+
+import { Button } from 'semantic-ui-react'
+
+const CLAIM_FILES_QUERY = gql`
+  query ClaimFilesQuery($id: ID!) {
+    claim(id: $id) {
+      claimFiles {
+        claimFileId
+        fileUploadUrl
+        markedAsDeleted
+        category
+      }
+    }
+  }
+`
+
+const MARK_CLAIM_FILE_AS_DELETED = gql`
+  mutation markClaimFileAsDeleted($claimId: ID!, $claimFileId: ID!) {
+    markClaimFileAsDeleted(claimId: $claimId, claimFileId: $claimFileId)
+  }
+`
+
+interface State {
+  confirmed: boolean
+}
+
+export class DeleteButton extends React.Component<
+  {
+    claimId: string
+    claimFileId: string
+    showNotification: (data: any) => void
+  },
+  State
+> {
+  public state = {
+    confirmed: false,
+  }
+
+  public render() {
+    return (
+      <Mutation
+        mutation={MARK_CLAIM_FILE_AS_DELETED}
+        refetchQueries={() => [
+          {
+            query: CLAIM_FILES_QUERY,
+            variables: {
+              id: this.props.claimId,
+            },
+          },
+        ]}
+      >
+        {(mutation, { loading }) => (
+          <>
+            {!this.state.confirmed ? (
+              <Button
+                onClick={(e) => {
+                  e.preventDefault()
+                  this.toggleConfirmed()
+                }}
+                disabled={loading}
+              >
+                Delete
+              </Button>
+            ) : (
+              <Button
+                disabled={loading}
+                onClick={() => this.handleClick(mutation)}
+              >
+                Are you sure?
+              </Button>
+            )}
+          </>
+        )}
+      </Mutation>
+    )
+  }
+
+  private handleClick = (mutation) => {
+    mutation({
+      variables: {
+        claimId: this.props.claimId,
+        claimFileId: this.props.claimFileId,
+      },
+    })
+      .then(() => {
+        this.props.showNotification({
+          message: 'file has been marked as deleted',
```

Comment:
From a UX perspective the file has been deleted I guess:
```suggestion
          message: 'File has been deleted',
```